components/ProfileForm.tsx

```diff
@@ -0,0 +1,61 @@
+'use client'
+
+import { useRef, useState } from 'react'
+import { z } from 'zod'
+
+import { CustomInput } from './atoms'
+
+//validation scheme
+const profileSchema = z.object({
```

Comment:
it's necessary to split this type of component into container component and view component, all validation (schemas) must be placed in separate files, you can put them in the types folder in the schemas.ts file